Src/Coravel/Scheduling/Schedule/Restrictions/CustomRestrictions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace Coravel.Scheduling.Schedule.Restrictions
+{
+    public class CustomRestrictions : IRestrictions
+    {
+        private Func<bool> func;
+
+        public CustomRestrictions(){
+
+        }
+
+        public CustomRestrictions(Func<bool> func)
+        {
+            this.func = func;
+        }
+
+        public void SetRestriction(Func<bool> func)
+        {
+            this.func = func;
+        }
+
+        public bool PassesRestrictions(DateTime utcNow)
+        {
+            return func?.Invoke() ?? true;
```

Comment:
I don't think Invoke() can ever return a null?

Reply:
The null check applies to the func? not for the Invoke()

Reply:
Sry, you're right.

If you go the route I mentioned in the other comment, it would be possible  for users to call `When` multiple times:

```
.When(..stuff)
.When(...stuff)
```

Maybe storing a List<Func<bool>> would work, and then expose a method to make sure they all pass?

Reply:
That was my original thought of using it at this point. I had the issue that the time restriction was returning null.

Reply:
scheduler.ScheduleAsync(async () =>
            {
                await Task.Delay(1);
                taskRunCount++;
            }).Where (() => taskRunCount == 2).EveryMinute();

You were right in wanted to keep the fluent nature of the API, my fault for not enforcing it before. Think this what you had in mind.